### PR TITLE
Backoff now only happens in proper status range

### LINKED ISSUE #34: Backoff should only happen on HTTP 500+ errors

### PR DESCRIPTION
**Pull request by SebastianNeiswanger**
closes #34

### CODE REVIEW

**Comment by SebastianNeiswanger** [comment]
Removed the test console.log so it should be good now

File: src/main.ts [hunk 1]
```diff
@@ -48,14 +48,16 @@ const waitForInput = () => {
...
-  } catch (error) {
-    await new Promise(f => setTimeout(f, backoff * 1000));
-    backoff *= 2
-    if (backoff > 128) { backoff = 128 }
-    sendCheckIn(checkin, backoff)
-  }
+  await axios.post('/checkin', checkin)
+  .catch (async function (error) {
+    const status = error.toJSON().status
+    if (status >= 500 || status === null) {
```
Comment (SpBills) [inline src/main.ts:71]: Hm. May prefer a try catch here rather than a `.catch` but this works for now.

File: src/main.ts [hunk 2]
```diff
@@ -48,14 +48,16 @@ const waitForInput = () => {
...
-    const result = await axios.post('/checkin', checkin);
-  } catch (error) {
-    await new Promise(f => setTimeout(f, backoff * 1000));
-    backoff *= 2
-    if (backoff > 128) { backoff = 128 }
-    sendCheckIn(checkin, backoff)
-  }
+  await axios.post('/checkin', checkin)
+  .catch (async function (error) {
+    const status = error.toJSON().status
```
Comment (SpBills) [inline src/main.ts:70]: This works? What? Cool.

**Review by SpBills — APPROVED** [review]
LGTM

**Pull request merged**